cmd/hook.go

```diff
@@ -669,7 +674,11 @@ Gitea or set your environment appropriately.`, "")
 		if err != nil {
 			return err
 		}
-		if rs.OldOID != git.EmptySHA {
+		hash, err := git.HashFromString(rs.OldOID)
```

Comment:
TBH, `hash` doesn't seem a good concept here.

Although I don't know how git calls it internally/officially, while I think Gitea also uses `commit id` for such type, which reads better than `hash`.


eg:

```
var commitID git.CommitID
git.IsCommitIDEmpty(...)
git.CommitIDFromString(...)
```

Reply:
They have renamed Sha => OID. We could follow similar  naming scheme but let's not change it again. I would argue that CommitID is worse than Hash since it doesn't necessarily reflect a commit id, for example tree id and similar, or the special empty tree

Hash => OID to follow upstream?

Reply:
Hmm .... I was thinking about these "SHA" related code was all related to "commit".

If we'd like to reflect git objects, maybe `ObjectID` is a better name?

Reply:
We also have OID in places in gitea code and git uses OID everywhere. If we change this, I think it would be better to be consistent and have types `OID` and `OIDType`

@6543 @lunny any preferences here?

Reply:
`OIDType` also looks good to me.

Reply:
Maybe `ObjectFormat` is the best name, ref `https://git-scm.com/docs/git-init`.

```
--object-format=<format>

    Specify the given object format (hash algorithm) for the repository. The valid values are sha1 and (if enabled) sha256. sha1 is the default.
```

Reply:
So, now he have 3 people and 4 opinions about naming things... Maybe we can leave it for now as-is and maybe change it later, if at all, toward the end of the review process.

Reply:
`ObjectFormat` is not a right name. It implies that the value should be SHA1 or SHA256. But here we need to store the value for Object ID.

Reply:
For any type changes, see the `hash.go`. We have now two types there `Hash`  and `HashType`. If we want to change this to `OID` and `OIDType`, that's fine, but I will not want to change it back later since all these renames is actually a lot of work! (for little gain :)

Reply:
so ...
... we dont have to have the same naming schema as internal `git` code. As to us this is a dependency.

I'm in fafour of HashType and Hash ... CommitID and ID.

Reply:
Maybe use `ObjectFormat` instead of `HashType`, `ObjectID` instead of `Hash`?